pandas/tests/frame/test_combine_concat.py

```diff
@@ -937,6 +937,18 @@ def test_concat_datetime_datetime64_frame(self):
         # it works!
         pd.concat([df1, df2_obj])
 
+    def test_combine_first_empty_dataframe(self):
+        # #29562
+        df1 = pd.DataFrame(columns=["a"])
+        df2 = pd.DataFrame(columns=["b"])
+
+        exp = pd.DataFrame(columns=["a", "b"])
+        res = df1.combine_first(df2)
+
+        res_rev = df2.combine_first(df1)
+        tm.assert_frame_equal(res, exp)
```

Comment:
do these sequentially

call these res and exp

Reply:
@iamrajhans can you update this? i.e. do the first assert before doing the second combine_first.

Can you make sure we have tests for empty dataframes that have rows but no columns?